metadata-ingestion/tests/unit/reporting/test_ingestion_stage.py

```diff
@@ -74,3 +80,23 @@ def test_ingestion_stage_context_report_handles_nested_stages():
     inner_sum = inner1_duration + inner2_duration
     # generous tolerance to avoid flakiness in test
     assert outer_duration == pytest.approx(inner_sum, abs=0.05)
+
+
+def test_ingestion_high_stage_context_records_duration():
+    report = IngestionStageReport()
+    with report.new_high_stage(stage=IngestionHighStage.PROFILING):
+        time.sleep(0.1)
+    assert len(report.ingestion_high_stage_seconds) == 1
+    assert IngestionHighStage.PROFILING.value in report.ingestion_high_stage_seconds
+    assert report.ingestion_high_stage_seconds[IngestionHighStage.PROFILING.value] > 0
```

Comment:
we should also assert on `ingestion_stage_durations`, no?

Reply:
I only made changes to `ingestion_high_stage_seconds` so I am only adding tests for that.